pycbc/psd/analytical.py

```diff
@@ -96,6 +97,10 @@ def from_string(psd_name, length, delta_f, low_freq_cutoff):
         raise ValueError(psd_name + ' not found among analytical '
                          'PSD functions.')
 
+    if type(length) is not int:
```

Comment:
Does this work if it's a numpy.int32 for example?

Reply:
Maybe we don't care...

Reply:
With a 1-element int32 array it will still print the warning, but otherwise produce the correct result. An array longer than 1 will raise a TypeError.

Reply:
Hold on, there is a better way for the type check.